pedsnetdcc/tests/prepdb_test.py

```diff
@@ -0,0 +1,182 @@
+            with conn.cursor() as cursor:
+                for role in TestPrepareDatabase.roles:
+                    cursor.execute('CREATE ROLE {}'.format(role))
+        conn.close()
+
+    def setUp(self):
+        """Set up for test(s).
+
+        1) Set self.model_version
+        2) Set self.dbname based on the model_version
+        3) Set self.conn_str based on testing.postgresql test instance
+        4) Create test roles
+        :raise: ValueError
+        """
+        self.model_version = '2.2.0'
+        self.dbname = _make_database_name(self.model_version)
+
+        self.postgresql = Postgresql()  # See setupModule
+        self.dburi = self.postgresql.url()
+        self.conn_str = make_conn_str(self.dburi)
+
+        self.create_roles()
+
+    def tearDown(self):
+        """Clean up after tests.
+
+        Note that if an exception is raised in setUp, tearDown is not invoked.
+
+        Drop roles.
+        """
+        self.postgresql.stop()
+
+    def check(self, dcc_only=False):
+        """
+        1) Connect to the database that should have been created.
+        2) Get a list of schemas and compare
+
+        :param bool dcc_only: if True, check dcc-only scenario
+        """
+        conn_str = _conn_str_with_database(self.conn_str, self.dbname)
+
+        # Make a list of expected schemas
+        schema_list = []
+        schema_suffixes = ('achilles', 'harvest', 'pcornet', 'pedsnet',
+                           'id_maps')
+
+        for site in _sites_and_dcc(dcc_only=dcc_only):
+            for schema_suffix in schema_suffixes:
+                schema_list.append('{}_{}'.format(site, schema_suffix))
+        schema_list.append('vocabulary')
+        schema_list.append('public')
+        schema_list.extend(('vocabulary', 'public', 'dcc_ids'))
+
+        expected_schemas = set(schema_list)
+        actual_schemas = set(introspect_schemas(conn_str))
+        self.assertEqual(actual_schemas, expected_schemas)
+
+    def check_conn_str(self):
+        """Validate the connection string.
+
+        1) Make sure dbname is not in the test DB URL.
+        2) Make sure a connection can be established.
+
+        :return: None
+        :raise: psycopg2.OperationalError, ValueError
+        """
+        match = re.search(r'\bdbname=([^ ]+)', self.conn_str)
+        if match and match.group(1) == self.dbname:
+            raise ValueError(
+                'URL should not contain the database to be created')
+
+        # An exception is raised if a connection cannot be established
+        with psycopg2.connect(self.conn_str) as conn:
+            pass
+        conn.close()
+
+    def drop_database(self):
+        """Drop the test database"""
+        sql = 'drop database {}'.format(self.dbname)
+        Statement(sql).execute(self.conn_str)
+
+    def test_prep_db(self):
+        """Test for sites + dcc"""
+        prepare_database(self.model_version, self.conn_str)
+        self.check()
+        self.drop_database()
```

Comment:
No longer needed; I thought testing.postgresql was caching the cluster, but it is caching the pre-cluster initted files; returns a fresh cluster each time.